mediator/src/controllers/service-request.ts

```diff
@@ -19,18 +22,40 @@ async function createServiceRequest(request: ServiceRequest) {
       }
     };
 
-    const res = await axios.get(`${fhirUrl}/Patient/${patientId}`, options);
+    const res = await axios.get(`${fhirUrl}/Patient/?identifier=${patientId}`, options);
 
     if (res.status !== 200) {
       return {status: res.status, data: res.data};
     }
 
-    // todo => @njogz to add task creation process.
+    // generate subscription resource
+    const FHIRSubscriptionResource = generateFHIRSubscriptionResource(patientId, callback_url);
+    const subscriptionRes = await axios.post(`${fhirUrl}/Subscription`, FHIRSubscriptionResource, options);
 
-    return {status: res.status, data: res.data};
+    if (subscriptionRes.status !== 201) {
+      return {status: subscriptionRes.status, data: subscriptionRes.data};
+    }
+
+    // call the CHT API to set up the follow up task
+    const chtApiUrl = generateApiUrl(CHT.url, CHT.username, CHT.password);
+    const chtRes = await axios.post(chtApiUrl, {
+      _meta: {
+        form: "interop_follow_up",
+      },
+      patient_uuid: patientId,
+    });
+
+    if (chtRes.data.success !== true) {
+      // TODO: delete the subscription
```

Comment:
Do we have an issue to tackle this `TODO`? If not, I recommend opening one (+ explaining why we need to delete the subscription). There is no such step in the sequence diagram, and it's worth a discussion on wether we need to delete it or not.